File: Tests/executors/CSharp/CoreExecutors/HelloExecutor.cs
using Framework;

public class HelloExecutor : StatelessExecutor
{
	private int value;

	public override void Init(ExecutorSettings executorSettings)
	{
		value = (int)executorSettings.GetInitParameters()["number"];
	}

	public override void DoGet(HttpRequest request, HttpResponse response)
	{
		response.SetBody
		(
			new
			{
				message = "Hello, World!",
				number = value
			}
		);
	}

	public override void DoPost(HttpRequest request, HttpResponse response)
	{
		DoGet(request, response);
	}

	public override void DoHead(HttpRequest request, HttpResponse response)
	{
		DoGet(request, response);
	}

	public override void DoPut(HttpRequest request, HttpResponse response)
	{
		DoGet(request, response);
	}

	public override void DoDelete(HttpRequest request, HttpResponse response)
	{
		DoGet(request, response);
	}

	public override void DoPatch(HttpRequest request, HttpResponse response)
	{
		DoGet(request, response);
	}

	public override void DoConnect(HttpRequest request, HttpResponse response)
	{
		DoGet(request, response);
	}
}

File: Tests/executors/CXX/core/src/HelloExecutor.cpp
#include "HelloExecutor.h"

void HelloExecutor::init(const framework::utility::ExecutorSettings& executorSettings)
{
	framework::JsonParser initParameters = executorSettings.getInitParameters();

	value = initParameters.get<int64_t>("number");
}

void HelloExecutor::doGet(framework::HttpRequest& request, framework::HttpResponse& response)
{
	framework::JsonBuilder builder;

	builder["message"] = "Hello, World!";
	builder["number"] = value;

	response.setBody(builder);
}

void HelloExecutor::doPost(framework::HttpRequest& request, framework::HttpResponse& response)
{
	this->doGet(request, response);
}

void HelloExecutor::doHead(framework::HttpRequest& request, framework::HttpResponse& response)
{
	this->doGet(request, response);
}

void HelloExecutor::doPut(framework::HttpRequest& request, framework::HttpResponse& response)
{
	this->doGet(request, response);
}

void HelloExecutor::doDelete(framework::HttpRequest& request, framework::HttpResponse& response)
{
	this->doGet(request, response);
}

void HelloExecutor::doPatch(framework::HttpRequest& request, framework::HttpResponse& response)
{
	this->doGet(request, response);
}

void HelloExecutor::doConnect(framework::HttpRequest& request, framework::HttpResponse& response)
{
	this->doGet(request, response);
}

DEFINE_EXECUTOR(HelloExecutor)
